pkg/kubectl/cmd/util/factory_client_access.go

```diff
@@ -233,6 +235,19 @@ func (f *ring0Factory) RESTClient() (*restclient.RESTClient, error) {
 	return restclient.RESTClientFor(clientConfig)
 }
 
+func (f *ring0Factory) ScaleClient(mapper meta.RESTMapper) (scaleclient.ScalesGetter, error) {
```

Comment:
move this out to the ring2Factory and it can be no-arg.  See the example here: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/util/factory_builder.go#L46